### PR TITLE
feat: set MAX VM cycles from cli

### LINKED ISSUE #821: How is `MaxCycles` set for gno.land?
This is not a bug, but a question - I'm wondering how the `MaxCycles` gets set and why it is what it is. It looks like its set to `10M` (https://github.com/gnolang/gno/blob/master/tm2/pkg/sdk/vm/keeper.go#L177), but what is the reasoning for that number? 

Practically speaking, I am running into a `CPU cycle overrun` error by exceeding `MaxCycles` with a pretty simple realm (https://github.com/schollz/gno/tree/bytebeat-works) that generates audio. The realm works for 1 second of audio, but 2 seconds seems to exceed the maximum cycles. The idea for this realm was to make an audio-based nft where the contract itself generates the audio. If this is out-of-bounds of what is considered for Gno, that's okay.

### PR DESCRIPTION
**Pull request by ajnavarro**
# Description

Before this change, some VM operations like loading a package or calling a public realm method had a hardcoded limit of 10M max allowed cycles. With this change that limit can be modified.

Usage: `gnoland --max-vm-cycles=0`

It closes #821 

## Contributors Checklist

- [X] Added new tests, or not needed, or not feasible
- [X] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [X] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [X] Added references to related issues and PRs
- [X] Provided any useful hints for running manual tests

### CODE REVIEW

**Review by moul — CHANGES_REQUESTED** [review]
Hey @ajnavarro, I agree that we should make this editable. However, there are a few things to keep in mind:

1. In the future, the value won't be dynamically determined by the node itself. Instead, it will be a variable derived from the genesis configuration. This variable can be updated through DAO votes in a new `r/system/variables` realm (#1856). You can find more information about this in the following link: [https://github.com/gnolang/gno/issues/761#issuecomment-1516632177](https://github.com/gnolang/gno/issues/761#issuecomment-1516632177).
2. Our goal is to simplify the configuration process by introducing variables that can be set in both config files and the command-line interface (CLI). This will eliminate the need for concurrent implementation of configuration. For more details, please refer to this pull request: [https://github.com/gnolang/gno/pull/729#pullrequestreview-1380137928](https://github.com/gnolang/gno/pull/729#pullrequestreview-1380137928). Additionally, you can explore issue #731 for further information.

Here's my suggestion for the next steps:

1. Wait for the completion of the merge for both #827 and #762 (which will likely happen today).
2. Once those are merged, proceed by updating the commands package to automatically register flags for available configuration files. You can find more details in issue #731.
3. Finally, add your new flag, but make sure to give it a name that reflects its purpose as the `GenesisMaxVMCycle` rather than just the `MAX_VM_CYCLE`.

File: gno.land/cmd/gnoland/main.go
Comment (thehowl) [inline gno.land/cmd/gnoland/main.go:113]: just a small suggestion :)

```suggestion
		10_000_000,
```

**Comment by jaekwon** [comment]
reading now

File: tm2/pkg/sdk/vm/keeper.go
Comment (jaekwon) [inline tm2/pkg/sdk/vm/keeper.go:55]: eventually we should pull out option fields into a VMOptions or whatever convention this sdk already uses.

Comment (jaekwon) [reply tm2/pkg/sdk/vm/keeper.go:55]: maybe worth doing now, or at least putting a comment in for the next person.

**Review by jaekwon — APPROVED** [review]
pre-approving besides existing comments.

File: gno.land/cmd/gnoland/main.go
Comment (ajnavarro) [reply gno.land/cmd/gnoland/main.go:113]: TIL

File: tm2/pkg/sdk/vm/keeper.go
Comment (ajnavarro) [reply tm2/pkg/sdk/vm/keeper.go:55]: I am adding a TODO for now. I can handle it on a different PR if it is a priority.

**Comment by ajnavarro** [comment]
@moul addressed all your comments!

**Review by moul — APPROVED** [review]

File: tm2/pkg/sdk/vm/keeper.go
Comment (moul) [reply tm2/pkg/sdk/vm/keeper.go:55]: By the way, I'm not perturbed (yet) by the number of parameters, and I appreciate the fact that the compiler will raise an error if we add or remove even a single parameter with the current implementation.

**Comment by jaekwon** [comment]
Thank you Antonio!

**Pull request merged**
